File: compare.lua
-- who doesn't love some fibonnaci numbers

local function fibonacci(n) return n<2 and n or fibonacci(n-1)+fibonacci(n-2) end

local time = os.clock()
local fibs = {}
for i=1,30 do 
	fibs[i] = fibonacci(i)
end
print("Lua finished in: "..1000*(os.clock()-time).." milliseconds")

File: compare.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int fibonacci(int n) { if (n < 2) return n; return fibonacci(n-1) + fibonacci(n-2); }

int main() { 
    clock_t sStart = clock();
    int *fibs = (int *)malloc(30*sizeof(int));
    for (int i = 0; i < 30; i++) {
        fibs[i] = fibonacci(i);
    }
    free(fibs);
    clock_t sEnd = clock();
    printf("C finished in: %g milliseconds", 1000*(double)(sEnd-sStart)/CLOCKS_PER_SEC);
    return 0;
}
